app/test/unit/services/subscription.service.test.js

```diff
@@ -0,0 +1,57 @@
+const chai = require('chai');
+const SubscriptionsService = require('services/subscription.service');
+
+const { expect } = chai;
+
+describe('Subscription Service', () => {
+    const mockArea = {
+        application: 'gfw',
+        env: 'production',
+        tags: [],
+        status: 'saved',
+        public: false,
+        fireAlerts: true,
+        deforestationAlerts: true,
+        webhookUrl: '',
+        monthlySummary: false,
+        subscriptionId: '',
+        email: 'test.email@wri.org',
+        language: 'en',
+        _id: '7604df303569488fbf928561',
+        name: 'Kiambu, Kenya',
+        geostore: '33b01a49bf9b56a8b56ce042a24f6567',
+        wdpaid: null,
+        userId: 'user123',
+        admin: { adm0: 'KEN', adm1: '13' },
+        datasets: [],
+        image: '',
+        deforestationAlertsType: 'glad-all',
+        createdAt: '2025-01-21T14:20:05.453Z',
+        updatedAt: '2025-01-21T14:20:05.453Z',
+        adminVersions: [
+            {
+                provider: 'gadm',
+                version: '3.6',
+                geostore: '33b01a49bf9b56a8b56ce042a24f6567',
+                country: {},
+                region: {}
+            }
+        ],
+        __v: 0
+    };
+
+    it('should construct the correct subscription body for a given area', () => {
+        const result = SubscriptionsService.getRequestBodyForSubscriptionFromArea(mockArea);
+
+        expect(result).to.be.an('object');
+        expect(result.name).to.equal('Kiambu, Kenya');
+        expect(result.language).to.equal('en');
+        expect(result.datasets).to.deep.equal(['glad-all', 'viirs-active-fires']);
+        expect(result.userId).to.equal('user123');
+        expect(result.params).to.deep.equal({
+            iso: { country: 'KEN', region: '13' },
+            area: '7604df303569488fbf928561',
+        });
```

Comment:
The bummer about the multiple `expect`s is that the test stops at the first failure. So, you can't see how many properties are actually failing.

I _think_ doing one `deep.include` will give you nice output if it fails:
```suggestion
        expect(result).to.deep.include({
          name: 'Kiambu, Kenya',
          language: 'en',
          datasets: [
            'glad-all',
            'viirs-active-fires'
          ],
          userId: 'user123',
          params: {
            iso: {
              country: 'KEN',
              region: '13',
            },
            area: '7604df303569488fbf928561',
          }
        });
```

Reply:
Thanks @gtempus, updated accordingly. I had to split dataset property test to a new case to make deep.include happy.

Reply:
Ah, okay. Still, I think that's better than the multiple `expect`s.